community/kernel/src/main/java/org/neo4j/kernel/impl/api/DefaultTransactionTracer.java

```diff
@@ -32,21 +33,37 @@
 import org.neo4j.kernel.impl.transaction.tracing.StoreApplyEvent;
 import org.neo4j.kernel.impl.transaction.tracing.TransactionEvent;
 import org.neo4j.kernel.impl.transaction.tracing.TransactionTracer;
+import org.neo4j.kernel.impl.util.JobScheduler;
 
-public class DefaultTransactionTracer implements TransactionTracer, LogRotationMonitor
+public class DefaultTransactionTracer implements TransactionTracer, LogRotationMonitor, LogRotationEventObservable
 {
     private final Clock clock;
+    private final JobScheduler jobScheduler;
+
     private final AtomicLong counter = new AtomicLong();
     private final AtomicLong accumulatedTotalTimeNanos = new AtomicLong();
 
+    private volatile Observer observer;
     private long startTimeNanos;
 
+    private final BiFunction<Long,Long,Runnable> notifyObserver = ( Long delta, Long count ) -> () -> {
```

Comment:
I understand that need but it's not what i'm trying to influence. I mean its reads really bad.